### PR TITLE
Adds validation webhook for EdgeConnect CR

### PR DESCRIPTION
**Pull request by aorcholski**
## Description
Adds new EdgeConnect endpoint to validation webhooks:
- adds new entry to validation webhook configuration 
- `/validate-edgeconnect` endpoint is registered  in the webhook manager

Adds EdgeConnect checks:
- apiServer
  - has proper suffix
  - has tenant field

## How can this be tested?
Try to install invalid EdgeConnect CR having:
- invalid apiServer, for example  `example.apps.com`
- invalid apiServer, for example  `.apps.dynatrace.com`
```
# kubectl -n dynatrace apply -f - <<EOF
apiVersion: dynatrace.com/v1alpha1
kind: EdgeConnect
metadata:
  name: edgeconnect
  namespace: dynatrace
spec:
  apiServer: "example.apps.com"

  oauth:
    clientSecret: "secret"
    endpoint: "endpoint"
    resource: "urn"
EOF
```    

## Checklist

- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly with a single label
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: src/webhook/validation/edgeconnect/validation.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+func newEdgeConnectValidator(apiReader client.Reader, cfg *rest.Config) admission.Handler {
+	return &edgeconnectValidator{
+		apiReader: apiReader,
+		cfg:       cfg,
+	}
+}
+
+func AddEdgeConnectValidationWebhookToManager(manager ctrl.Manager) error {
+	manager.GetWebhookServer().Register("/validate-edgeconnect", &webhook.Admission{
```
Comment (waodim) [inline src/webhook/validation/edgeconnect/validation.go:32]: Do we want to adapt the path of the dynakube validator to `"/validate-dynakube"`? 

FYI: Apart from this question I would approve this PR as I tested it and it works.

File: config/helm/chart/default/templates/Common/webhook/validatingwebhookconfiguration.yaml [hunk 2]
```diff
@@ -42,4 +42,26 @@ webhooks:
...
     sideEffects: None
+  - admissionReviewVersions:
+      - v1
+      - v1beta1
+      - v1alpha1
+    clientConfig:
+      service:
+        name: dynatrace-webhook
+        namespace: {{ .Release.Namespace }}
+        path: /validate-edgeconnect
```
Comment (andriisoldatenko) [inline config/helm/chart/default/templates/Common/webhook/validatingwebhookconfiguration.yaml:53]:
```suggestion
        path: /validate/edgeconnect
```
nitpicking :)

File: src/webhook/validation/edgeconnect/validation.go (hunk 1 above)
Comment (aorcholski) [reply src/webhook/validation/edgeconnect/validation.go]: fixed

**Review by andriisoldatenko — COMMENTED** [review]
code LGTM

**Review by waodim — APPROVED** [review]
Tested it again - still works and LGTM!

**Review by andriisoldatenko — APPROVED** [review]

**Pull request merged**